internal/hcp/registry/hcl.go

```diff
@@ -140,12 +140,14 @@ func NewHCLRegistry(config *hcl2template.PackerConfig, ui sdkpacker.Ui) (*HCLReg
 		diags = append(diags, bucketDiags...)
 	}
 
-	if diags.HasErrors() {
-		return nil, diags
+	registeringDiags := registerAllBuilds(&config.Builds, bucket)
```

Comment:
I know this will change, but in the current state, since `config.Builds[0]` is where we get the registry information from, this seems a bit weird to me, but I don't think it's necessarily wrong to do it now, just that it feels a bit off when looking at the code.
I wonder if we can add a small comment before doing this so as to clarify that the logic will be updated in the not too distant future, so that we have a trace in the code?